yt_idv/scene_data/block_collection.py

```diff
@@ -95,6 +134,15 @@ def add_data(self, field, no_ghost=False):
             VertexAttribute(name="in_right_edge", data=re)
         )
 
+        phi_plane_re = np.array(phi_plane_re, dtype="f4")
+        phi_plane_le = np.array(phi_plane_le, dtype="f4")
+        self.vertex_array.attributes.append(
+            VertexAttribute(name="phi_plane_le", data=phi_plane_le)
+        )
+        self.vertex_array.attributes.append(
+            VertexAttribute(name="phi_plane_re", data=phi_plane_re)
+        )
+
```

Comment:
So it occurs to me that we should have the cartesian volume rendering as the basic option, and we can/should have spherical as a subclass.  Let's not try cramming it all in; @neutrinoceros 's work on having things be visible in index-space coordinates is encouraging me to think of it in both ways.

That's not really specific *here* though, except that what we may want to do is to have this be a supplemental function that gets called if the *component* accessing the data is viewing it in spherical coordinates.

Reply:
ya! I agree totally! I just jammed it in here to get things in initially working. Unjamming it and subclassing it relates to your vectorization question I think. If I vectorize that function, it'll be easier to have a sublcass that takes the output from the standard cartesian version to calculate the new spherical-only attributes.

Reply:
latest push no longer breaks things for cartesian coords (it's now vectorized and only runs if spherical). 

I still want to keep thinking on how to better capture the different geometries though. I initially thought it may make sense to have a `SphericalBlockCollection`, but implementing it wasn't super straightforward because the calculations rely on the left and right edges, which currently only get saved via the `vertex_array` attribute **after** casting to "f4". So a standard construction like:

```python
class SphericalBlockCollection(BlockCollection):
        def add_data(self, field, no_ghost=False):
            super().add_data(field, no_ghost=no_ghost)
            
            << --- calculate phi-normal planes from left, right edges --- >>
```            
would require us also saving the full `le` and `re` arrays as attributes. Since they're only needed for an intermediate step, it seemed better to just add a geometry check to `BlockCollection.add_data()`. But i'm open to ideas here!